src/arch/arm/object/smc.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2021, DornerWorks Ltd.
+ *
+ * SPDX-License-Identifier: GPL-2.0-only
+ */
+#include <config.h>
+
+#ifdef CONFIG_ALLOW_SMC_CALLS
+#include <arch/object/smc.h>
+
+compile_assert(n_msgRegisters_less_than_smc_regs, n_msgRegisters <= NUM_SMC_REGS);
+
+static exception_t invokeSMCCall(word_t *buffer, bool_t call)
+{
+    word_t i;
+    seL4_Word arg[NUM_SMC_REGS];
+    word_t *ipcBuffer;
+
+    for (i = 0; i < NUM_SMC_REGS; i++) {
+        arg[i] = getSyscallArg(i, buffer);
+    }
+
+    ipcBuffer = lookupIPCBuffer(true, NODE_STATE(ksCurThread));
+
+    register seL4_Word r0 asm("x0") = arg[0];
```

Comment:
Maybe this should have a compile time assert that `NUM_SMC_REGS` is at least 8. It's unlikey to ever change, but may catch cases where somebody is experimenting. I'd expect the compiler to issue a out-of-warning in this case automatically, though - but does this also break the build with the current defaults?

Reply:
If you want to make that in a separate PR then I'd be happy to accept it.

Reply:
will do